iosApp/iosApp/TestbedViewController.swift

```diff
@@ -138,8 +155,23 @@ class TestbedViewController: UIViewController {
         messenger.eventSubject
             .sink(receiveValue: updateWithEvent)
             .store(in: &cancellables)
+            authStateView.text = "Auth State: \(authState)"
     }
-    
+
+    private func signIn() {
```

Comment:
I think you could also change `buildOktaAuthorizeUrl() -> String?` to `buildOktaAuthorizeUrl() -> URL?` (doing the `URL(string: authUrlString!)` URL creating inside the builder func) and then simplify this `signIn` func logic to:

```
    private func signIn() {
        guard let authUrl = buildOktaAuthorizeUrl() else {
            authState = AuthState.error(errorCode: ErrorCode.AuthFailed.shared, message: "Failed to build Okta authorize URL.", correctiveAction: CorrectiveAction.ReAuthenticate.shared)
            updateAuthStateView()
            return
        }
        
        if UIApplication.shared.canOpenURL(authUrl) {
            UIApplication.shared.open(authUrl)
        }
    }
```

Reply:
Done